src/measure_theory/function/uniform_integrable.lean

```diff
@@ -0,0 +1,567 @@
+which estabishes a relation between uniform integrability, convergence in measure and
+Lp convergence.
+
+Uniform integrability plays a vital role in the theory of martingales most notably is used to
+fomulate the martingale convergence theorem.
+
+## Main definitions
+
+* `measure_theory.unif_integrable`: uniform integrability in the measure theory sense.
+* `measure_theory.uniform_integrable`: uniform integrability in the probability theory sense.
+
+# Main results
+
+* `measure_theory.unif_integrable_fintype`: a finite sequence of Lp functions is uniformly
+  integrable.
+* `measure_theory.tendsto_Lp_of_tendsto_ae`: a sequence of Lp functions which is uniformly
+  integrable converges in Lp if they converge almost everywhere.
+* `measure_theory.tendsto_in_measure_iff_tendsto_Lp`: Vitali convergence theorem:
+  a sequence of Lp functions converges in Lp if and only if it is uniformly integrable
+  and converges in measure.
+
+## Tags
+uniform integrable, uniformly absolutely continuous integral, Vitali convergence theorem
+-/
+
+noncomputable theory
+open_locale classical measure_theory nnreal ennreal topological_space
+
+namespace measure_theory
+
+open set filter topological_space
+
+variables {α β ι : Type*} {m : measurable_space α} {μ : measure α} [normed_group β]
+
+/-- Uniform integrability in the measure theory sense.
+
+A sequence of functions `f` is said to be uniformly integrable if for all `ε > 0`, there exists
+some `δ > 0` such that for all sets `s` with measure less than `δ`, the Lp-norm of `f i`
+restricted on `s` is less than `ε`.
+
+Uniform integrablility is also known as uniformly absolutely continuous integrals. -/
+def unif_integrable {m : measurable_space α} (f : ι → α → β) (p : ℝ≥0∞) (μ : measure α) : Prop :=
+∀ ⦃ε : ℝ⦄ (hε : 0 < ε), ∃ (δ : ℝ) (hδ : 0 < δ), ∀ i s, measurable_set s → μ s ≤ ennreal.of_real δ →
+snorm (s.indicator (f i)) p μ ≤ ennreal.of_real ε
+
+/-- In probability theory, a family of functions is uniformly integrable if it is uniformly
+integrable in the measure theory sense and is uniformly bounded. -/
+def uniform_integrable {m : measurable_space α} [measurable_space β]
+  (μ : measure α) (f : ι → α → β) (p : ℝ≥0∞) : Prop :=
+(∀ i, measurable (f i)) ∧ unif_integrable f p μ ∧ ∃ C : ℝ≥0, ∀ i, snorm (f i) p μ ≤ C
+
+lemma uniform_integrable.measurable [measurable_space β] {f : ι → α → β} {p : ℝ≥0∞}
+  (hf : uniform_integrable μ f p) (i : ι) : measurable (f i) :=
+hf.1 i
+
+lemma uniform_integrable.unif_integrable [measurable_space β] {f : ι → α → β} {p : ℝ≥0∞}
+  (hf : uniform_integrable μ f p) : unif_integrable f p μ :=
+hf.2.1
+
+lemma uniform_integrable.mem_ℒp [measurable_space β] {f : ι → α → β} {p : ℝ≥0∞}
+  (hf : uniform_integrable μ f p) (i : ι) :
+  mem_ℒp (f i) p μ :=
+⟨(hf.1 i).ae_measurable, let ⟨_, _, hC⟩ := hf.2 in lt_of_le_of_lt (hC i) ennreal.coe_lt_top⟩
+
+section unif_integrable
+
+/- This section deals with uniform integrability in the measure theory sense. -/
```

Comment:
You should use `/-! -/` to make that comment appear in the doc, and maybe add a section title
```suggestion
/-! ### `unif_integrable`

This section deals with uniform integrability in the measure theory sense. -/
```